modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/config/ClientConnFactoryBuilder.java

```diff
@@ -317,5 +321,46 @@ public ClientConnFactory createConnFactory(final HttpParams params) {
             return new ClientConnFactory(params);
         }
     }
-    
+
+    /**
+     *
+     * @return
+     */
+    public TransportOutDescription loadDynamicSSLConfig (TransportOutDescription transportOut) {
+
+        Parameter profilePathParam = transportOut.getParameter("dynamicSSLProfilesConfig");
+
+        //No Separate configuration file configured. Therefore using Axis2 Configuration
+        if (profilePathParam == null) {
+            return transportOut;
+        }
+
+        //Using separate SSL Profile configuration file, ignore Axis2 configurations
+        OMElement pathEl = profilePathParam.getParameterElement();
+        String path = pathEl.getFirstChildWithName(new QName("filePath")).getText();
+
+        try {
+            if (path != null) {
+
+                String separator = path.startsWith(System.getProperty("file.separator")) ?
+                                   "" : System.getProperty("file.separator");
+                String fullPath = System.getProperty("user.dir") + separator + path;
+
+                OMElement profileEl = new StAXOMBuilder(fullPath).getDocumentElement();
+                Parameter profileParam = new Parameter();
+                profileParam.setParameterElement(profileEl);
+                profileParam.setName("customSSLProfiles");
+                profileParam.setValue(profileEl);
+
+                transportOut.addParameter(profileParam);
+                log.info("customSSLProfiles configuration is loaded from path: " + fullPath);
+
+                return transportOut;
+            }
+
+        } catch (Exception e) {
```

Comment:
If generic exceptions were caught intentionally, might be a good idea to add a comment explaining why exactly this was done.